clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -138,7 +138,7 @@ public synchronized boolean updateRequested() {
     /**
      * Wait for metadata update until the current version is larger than the last version we know of
      */
-    public synchronized void awaitUpdate(final int lastVersion, final long maxWaitMs) throws InterruptedException {
+    public synchronized Cluster awaitUpdate(final int lastVersion, final long maxWaitMs) throws InterruptedException {
```

Comment:
Do we really need this change? If we immediately call `fetch` after `awaitUpdate`, we may get a subsequent version, but that shouldn't matter, right? The reason I ask is that it seems a bit arbitrary to return `Cluster` as opposed to other state held by `Metadata`.